pkg/collectors/builder.go

```diff
@@ -212,6 +216,57 @@ func (b *Builder) buildGroupCollector() *collector.Collector {
 	return collector.NewCollector(store)
 }
 
+func (b *Builder) buildInfrastructureCollector() *collector.Collector {
+	filteredMetricFamilies := metric.FilterMetricFamilies(b.whiteBlackList, infrastructureMetricFamilies)
+	composedMetricGenFuncs := metric.ComposeMetricGenFuncs(filteredMetricFamilies)
+
+	familyHeaders := metric.ExtractMetricFamilyHeaders(filteredMetricFamilies)
+
+	store := metricsstore.NewMetricsStore(
+		familyHeaders,
+		composedMetricGenFuncs,
+	)
+
+	reflectorPerNamespace(b.ctx, &configv1.Infrastructure{}, store,
```

Comment:
because these types (infrastructure, proxy, featuregate) are not namespaced, you don't need to create a reflector per namespace. You should likely create a function that just returns a reflector and takes no namespaces.